File: backend/src/main/java/backend/config/ProxyTargetConfig.java
package backend.config;

import java.util.List;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

import lombok.Data;

/** configuration for API proxy (loaded from application.yaml) */
@Component
@ConfigurationProperties(prefix="proxy")
@Data
public class ProxyTargetConfig {
	List<Target> targetList;

	@Data public static class Target {
		private String name;
		private String scheme = "http";
		private String host;
		private int port = 80;
		private boolean portForwarding = false;
		private Client client;

	}

	@Data public static class Client {
		private String id;
		private String secret;
	}
}

File: backend/src/main/kotlin/backend/config/ProxyTargetConfig.kt
package backend.config

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component

/** configuration for API proxy (loaded from application.yaml)  */
@Component
@ConfigurationProperties(prefix = "proxy")
data class ProxyTargetConfig (
    var targetList: List<Target>? = null
) {
    data class Target (
        var name: String? = null,
        var scheme: String = "http",
        var host: String? = null,
        var port: Int = 80,
        var portForwarding: Boolean = false,
        var client: Client? = null
    )

    data class Client (
        var id: String? = null,
        var secret: String? = null
    )
}